src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -1659,11 +2115,793 @@ End Class
+  IL_0140:  stelem.ref
+  IL_0141:  call       ""Function System.Linq.Expressions.Expression.Call(System.Linq.Expressions.Expression, System.Reflection.MethodInfo, ParamArray System.Linq.Expressions.Expression()) As System.Linq.Expressions.MethodCallExpression""
+  IL_0146:  ldtoken    ""System.Func(Of String)""
+  IL_014b:  call       ""Function System.Type.GetTypeFromHandle(System.RuntimeTypeHandle) As System.Type""
+  IL_0150:  call       ""Function System.Linq.Expressions.Expression.Convert(System.Linq.Expressions.Expression, System.Type) As System.Linq.Expressions.UnaryExpression""
+  IL_0155:  ldc.i4.1
+  IL_0156:  newarr     ""System.Linq.Expressions.ParameterExpression""
+  IL_015b:  dup
+  IL_015c:  ldc.i4.0
+  IL_015d:  ldloc.s    V_15
+  IL_015f:  stelem.ref
+  IL_0160:  call       ""Function System.Linq.Expressions.Expression.Lambda(Of System.Func(Of Derived, System.Func(Of String)))(System.Linq.Expressions.Expression, ParamArray System.Linq.Expressions.ParameterExpression()) As System.Linq.Expressions.Expression(Of System.Func(Of Derived, System.Func(Of String)))""
+  IL_0165:  stloc.s    V_14
+  // sequence point: End Sub
+  IL_0167:  ret
+}
+");
+        }
+
+        [Fact]
+        public void BinaryCompatibility_01()
+        {
+            var s0 = @"
+public class Base
+{
+    public virtual object M() => null;
+}
+";
+            var ref0 = CreateCompilationWithoutCovariantReturns(s0).EmitToImageReference();
+
+            var s1a = @"
+public class Mid : Base
+{
+}
+";
+            var ref1a = CreateCompilationWithoutCovariantReturns(s1a, references: new[] { ref0 }, assemblyName: "ref1").EmitToImageReference();
+
+            var s1b = @"
+public class Mid : Base
+{
+    public override string M() => null;
+}
+";
+            var ref1b = CreateCompilationWithCovariantReturns(s1b, references: new[] { ref0 }, assemblyName: "ref1").EmitToImageReference();
+
+            var s2 = @"
+public class Derived : Mid
+{
+    public override string M() => null;
+}
+";
+            var comp = CreateCompilationWithoutCovariantReturns(s2, references: new[] { ref0, ref1a }).VerifyDiagnostics(
+                // (4,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(4, 28)
+                );
+            verify1(comp);
+
+            comp = CreateCompilationWithCovariantReturns(s2, references: new[] { ref0, ref1a }).VerifyDiagnostics(
+                );
+            verify1(comp);
+            verify1(CompilationReferenceView(comp, ref0, ref1a));
+            verify1(MetadataView(comp, ref0, ref1a));
+            verify1(RetargetedView(comp, ref0, ref1a));
+
+            verify2(CompilationReferenceView(comp, ref0, ref1b));
```

Comment:
>CompilationReferenceView [](start = 20, length = 24)

Isn't this going to create a retargeting assembly?

Reply:
Perhaps the helper should assert that we are indeed reuse SourceAssemblySymbol

---
In reply to: [424776007](https://github.com/dotnet/roslyn/pull/44025#discussion_r424776007) [](ancestors = 424776007)